src/lib/sys.c

```diff
@@ -93,12 +91,16 @@ int run_command_full_params(const char *stdout_file, const char *stderr_file, ch
 	}
 	close(null_fd);
 
-	if (replace_fd(STDOUT_FILENO, stdout_file) < 0) {
-		goto error_child;
+	if (stdout_file) {
```

Comment:
I think you want the opposite. If (!stdout_file) we are printing to stdout, so we want to redirect.

Reply:
So if you check the code,

a.) if stdout_file arg is null, replace_fds returns a zero which it will print to STDOUT.
b.) if stdout_file arg is not null, replace_fds returns a replaced fd which points to a file

So I feel it is correct.

Reply:
Done